x-pack/plugins/endpoint/server/routes/alerts/details/handlers.ts

```diff
@@ -22,22 +22,29 @@ export const alertDetailsHandlerWrapper = function(
     try {
       const alertId = req.params.id;
       const response = (await ctx.core.elasticsearch.dataClient.callAsCurrentUser('get', {
+        // Remove the hard coded reference to an alert index here
+        // https://github.com/elastic/endpoint-app-team/issues/311
         index: EndpointAppConstants.ALERT_INDEX_NAME,
         id: alertId,
       })) as GetResponse<AlertEvent>;
 
+      const indexPattern = await endpointAppContext.indexPatternRetriever.getEventIndexPattern(ctx);
+
       const config = await endpointAppContext.config();
       const pagination: AlertDetailsPagination = new AlertDetailsPagination(
         config,
         ctx,
         req.params,
-        response
+        response,
+        indexPattern
       );
 
-      const currentHostInfo = await getHostData(ctx, response._source.host.id);
+      const currentHostInfo = await getHostData(ctx, response._source.host.id, indexPattern);
 
       return res.ok({
         body: {
+          // base64 encode the index in the response
+          // https://github.com/elastic/endpoint-app-team/issues/311
```

Comment:
would you please remove this comment and instead link to this line of code in the issue.